src/printer/smt2/smt2_printer.cpp

```diff
@@ -671,6 +671,38 @@ bool Smt2Printer::toStreamBase(std::ostream& out,
       toStream(out, hoa, lbind, toDepth);
       return true;
     }
+    else if (n.getOperator().getKind() == Kind::SKOLEM)
+    {
+      SkolemManager* sm = nm->getSkolemManager();
+      SkolemId id;
+      Node cacheVal;
+      if (sm->isSkolemFunction(n.getOperator(), id, cacheVal))
+      {
+        if (n.getNumChildren() != 0)
+        {
+          out << '(';
+        }
+        // FIXME: refactor to reuse the code for printing standalone skolems
```

Comment:
Remember to address this by refactoring this code in a common utility used in 2 places in this class.